fluent/src/fallback.js

```diff
@@ -71,3 +70,36 @@ function getContextForId(iterable, id) {
 
   return null;
 }
+
+/*
+ * Asynchronously map an identifier or an array of identifiers to the best
+ * `MessageContext` instance(s).
+ *
+ * @param {AsyncIterable} iterable
+ * @param {string|Array<string>} ids
+ * @returns {MessageContext|Array<MessageContext>}
+ */
+export async function mapContextAsync(iterable, ids) {
+  if (!Array.isArray(ids)) {
+    for await (const context of iterable) {
+      if (context.hasMessage(ids)) {
+        return context;
+      }
+    }
+  }
+
+  const foundContexts = new Array(ids.length).fill(null);
+
+  for await (const context of iterable) {
+    // XXX Switch to const [index, id] of id.entries() when we move to Babel 7.
+    // See https://github.com/babel/babel/issues/5880.
+    for (let index = 0; index < ids.length; index++) {
+      const id = ids[index];
+      if (!foundContexts[index] && context.hasMessage(id)) {
+        foundContexts[index] = context;
+      }
+    }
+  }
```

Comment:
Should there be an early exit from the loop in case all contexts are found?

Reply:
If I'm correct, please add a test that verifies that contexts are not unnecessarily resolved.

Reply:
Good point, thanks!